CenterForOpenScience/osf.io
# Feature/Node Delete Behavior Improvements [ENG-877]

## Purpose

Several items noticed lately around deleted nodes:
Restore node behavior is broken now that behavior introduced in December trashes the root folder, and restoring doesn't bring it back.  get_auth doesn't factor in whether or not a resource has been deleted. Old restored nodes need to be taken care of. 

## Changes
- Remove behavior that trashes root folder as part of node deletion process 
- Add migration management command that restores deleted osfstorage root folders - double checking in prod data that this is a small percentage, just introduced late last year). This should fix those restored nodes whose osfstorage addons appear to be disconnected.
- Make get_auth return a 404 if the node/preprint has been deleted

## QA Notes
Multiple things to check:
1) Create a node, add a file to it, note the file's guid - access the file in the API, and get the move link. Something like - https://files.us.staging.osf.io/v1/resources/jxfkt/providers/osfstorage/5c262827bf85ca001685385d/"  Delete the node.   Revisit that WB move link. Should get a 404 now instead of a 500.

2) Pre-management command check.  Delete some nodes. Restore them in the admin app. Note that the osfstorage addon appears not to load.  After we run the management command, these restored node's osfstorage addons should work again.

3) Post-management command.  Note that after deleting nodes and restoring them in the admin app, the osfstorage addon loads fine.

4) Light testing of general nodes delete behavior, and the items that should be affected - addons are removed
- Comment targets are updated
- node is marked as deleted
- subscriptions are removed
- logs are added
- search is updated
- DOI's updated
- files deleted

## Documentation



## Side Effects



## Deployment

There's a management command related to this ticket - `restore_deleted_root_folders`

## Ticket

https://openscience.atlassian.net/browse/ENG-877
